scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,496 @@
+
+
+def collect_as_cf(self: DataFrame,
+                  type: str = 'json') -> Tuple[List[Result], int, bool]:
+    """Collects the result of the DataFrame's execution plan as cloud fetch presigned URLs.
+
+    This method is a wrapper around the `to_cf` method of SparkConnectClient. It takes the
+    execution plan of the current DataFrame, converts it to a protocol buffer format, and then
+    uses the `to_cf` method to execute the plan and fetch results as presigned URLs.
+
+    Args:
+        type (str): The output format of the result, supported formats are 'json', 'csv', and 'arrow'.
+
+    Returns:
+        Tuple[List[Result], int, bool]: A tuple containing:
+            - A list of Result namedtuples, each containing a URL, row count, compressed size,
+              and uncompressed size of the part of the result.
+            - Total row count of all parts of the result.
+            - A boolean indicating whether the result is truncated or overflowed.
+    """
+    query = self._plan.to_proto(self._session.client)  # pyright: ignore
+    return self._session.client.to_cf(query, type)  # pyright: ignore
+
+
+DataFrame.collect_cf = collect_as_cf  # pyright: ignore
+
+
+def iterative_combine_jsons(json_directory: str, output_file: str) -> None:
+    """Combine jsonl files in json_directory into one big jsonl file.
+
+    This function does not work for nested subdirectories.
+
+    Args:
+        json_directory(str): directory containing the JSONL files
+        output_file(str): path to the output combined JSONL file
+    """
+    json_files = [f for f in os.listdir(json_directory) if f.endswith('.jsonl')]
+    with open(output_file, 'w') as outfile:
+        for file_name in json_files:
+            with open(os.path.join(json_directory, file_name), 'r') as infile:
+                for line in infile:
+                    outfile.write(line)
+    log.info('JSON files have been combined into a JSONL file.')
+
+
+def run_query(
+    query: str,
+    method: str,
+    cursor: Optional[Cursor] = None,
+    spark: Optional[SparkSession] = None,
+    collect: bool = True
+) -> Union[List[Row], DataFrame, SparkDataFrame]:
+    """Run SQL query via databricks-connect or databricks-sql.
+
+    Args:
+        query (str): sql query
+        method (str): select from dbsql and dbconnect
+        cursor (Optional[Cursor]): connection.cursor
+        spark (Optional[SparkSession]): spark session
+        collect (bool): whether to get the underlying data from spark dataframe
+    """
+    if method == 'dbsql':
+        if cursor is None:
+            raise ValueError(f'cursor cannot be None if using method dbsql')
+        cursor.execute(query)
+        if collect:
+            return cursor.fetchall()
+    elif method == 'dbconnect':
+        if spark == None:
+            raise ValueError(f'sparkSession is required for dbconnect')
+        df = spark.sql(query)
+        if collect:
+            return df.collect()
+        return df
```

Comment:
nit: there are a few instances of this `method == 'dbsql` v.s. `method == 'dbconnect'` in this file. A more Pythonic way to abstract this functionality is to create an abstract class that has these common methods and instantiate method-specific subclasses that inherit from the abstract class. The equivalent would be operating on an `ObjectStore` abstraction in [Composer](https://github.com/mosaicml/composer/blob/dev/composer/utils/object_store/object_store.py#L62C1-L62C28) but having specific `UCVolumeObjectStore` implementations

e.g.
```
class JsonToDeltaConverter(abc.ABC):
  def fetch_data():
    pass
  def run_query():
    pass
 
class DbConnectJsonToDeltaConverter(JsonToDeltaConverter):
  ...
class DbSqlJsonToDeltaConverter(JsonToDeltaConverter):
  ...
```

Reply:
Add the request to backlog ticket [JIRA](https://databricks.atlassian.net/issues/?jql=project+%3D+%22STR%22+ORDER+BY+created+DESC&atlOrigin=eyJpIjoiZWE2ZjNiMTAxMjg1NDdjMGFlNTIzNWQ0MjgzODI3YmMiLCJwIjoiaiJ9)